rust/ql/test/library-tests/controlflow/test.rs

```diff
@@ -141,6 +141,22 @@ mod if_expression {
         0
     }
 
+    fn test_and_if_let(a: bool, b: Option<bool>, c: bool) -> bool {
+        if a && let Some(d) = b {
+            d
+        } else {
+            false
+        }
+    }
+
+    fn test_and_if_let2(a: bool, b: i64, c: bool) -> bool {
+        if a && let d = b && c{
+            d > 0
+        } else {
+            false
+        }
+    }
+
```

Comment:
Let's perhaps do that in a follow-up PR?

Reply:
I don't think there's much value in adding partial support for `let` expressions without handling that case as well.

Is that change orthogonal to the one in this PR? If yes, then doing it in a follow up PR seems fine, otherwise I think I'd be easier to review the changes together.

Reply:
Yes, it's orthogonal to this PR; this PR merely improves CFG splitting a bit (see example in the PR description).

Reply:
Ok, great :)